deploy-board/deploy_board/webapp/group_view.py

```diff
@@ -1045,16 +1045,20 @@ def add_instance(request, group_name):
                 placement_group = params['placementGroup']
                 use_placement_group = True
 
-        # Check if asg is enabled and does not use placement group. Then launch in asg
-        if str(asg_status).upper() == "ENABLED" and not use_placement_group:
+        # When a group has an associated ASG, its status is either ENABLED or DISABLED.
+        # We should always launch in ASG when an ASG is set up regardless its status,
+        # unless placement group is specified.
+        if (asg_status == 'ENABLED' or asg_status == 'DISABLED') and not use_placement_group:
             launch_in_asg = True
 
         if launch_in_asg:
             # Launch hosts inside ASG / Bump ASG size by required instances
             autoscaling_groups_helper.launch_hosts(request, group_name, num, None)
```

Comment:
Below there is a message in the condition hosts are launched in the ASG, maybe the message should also apply here -- if so it could be moved out of the conditional logic and use host_ids as the test, or a flag: `content = '{} hosts have been launched to group {} (host ids: {})'.format(num, group_name, host_ids)` currently line 1081 in this diff.

Reply:
Unfortunately, `autoscaling_groups_helper.launch_hosts` doesn't return anything.